### ISSUE TITLE
loading video from FileReader

### ISSUE THREAD

**Issue by jsphweid** [processing/p5.js#1730]
--- tl:dr --- 
is there simply an easier way to load from `data:video/mp4;base64,AAAAGGZ0eXBtcDQyAAAAAGlzb21tcDQyAAAytG1vb3YAAABsbXZoZAAAAADUNY+x1DWPsQABX5AALq8FAAEAAAEAA`... etc. (e.target.result)

--- more ---

I'm trying to load a video from a drag and drop. Yes, I know P5JS already has one but I've ruled out using that because I'm doing this in angular2 / typescript with instance mode. Honestly I think I'm really close but I can't seem to finish.

I'll try to avoid the angular2 / typescript jargon.

Basically, I have a `p5file` that gets created before the sketch. It get's filled with basic info and when the data is processed by `FileReader()`, it get's filled, then the instance of p5js is created. In the setup() of that instance, I have `vid = p.createVideo([this.p5file]);`. Then the draw is executed as expected. It's looping like normal but `p.image(vid, 0, 0);` doesn't draw the mp4 as expected. I get no image. Slapping a debugger on there, it is evident that `vid` contains what you might expect (`t.MediaElement {elt: video, _pInst: undefined, _events: Object, width: 300, height: 150…}`... etc.) but no image appears. However, if I throw in something like `p.createVideo(['http://example.com/link/to/my/file.mp4'])` I get `t.MediaElement {elt: video, _pInst: undefined, _events: Object, width: 640, height: 480…}`. A different height / width.. and it's the same video file.

What is happening?

Here is some of my javascript/typescript code where I load the file.

```javascript
filesUpdated(files) { // from my angular2 dropzone
    var my = this;
    if (files.length === 1) {
        var reader = new FileReader();
        reader.onload = (function(theFile) { // go here
            let win: any = window; // get window (for .p5 object)
            my.flickerCycleService.p5file = new win.p5.File(theFile[0]); // extract most info
            return function(e) {
                my.flickerCycleService.p5file.data = e.target.result; // fill in actual data
                let win: any = window;
                new win.p5(my.flickerCycleService.sketch); // create sketch
            };
        })(files);
        reader.readAsDataURL(files[0]); // get first file of 1, i.e. [file]
    } else {
        alert("Only one file at a time. Refresh and try again.");
    }
}
```

EDITED - TL:DR at top...

**Comment by lmccart** [comment]
I just pushed some changes and tested with this example and it seems to work, so I think it should also work with the code you've posted.

```javascript
var input;
var img;

function setup() {
  input = createFileInput(handleFile);
  input.position(0, 0);
}

function draw() {
  if (img) {
    image(img, 0, 0, width, height);
  }
}

function handleFile(file) {
  print(file);
  if (file.type === 'image') {
    img = createImg(file.data);
  } else if (file.type === 'video') {
    img = createVideo('assets/'+file.name);
    img.loop();
  }
}
```

**Comment by jsphweid** [comment]
@lmccart Lauren, thanks for looking at this so quickly.

That may have fixed another issue but it didn't fix the one I was having trouble with, so I replicated the issue exactly in 60 lines in plain javascript in this plunker (with your latest p5js build).

https://plnkr.co/edit/NYDuICYK9DouVJ2fjzpe

Please have a look and let me know, thanks!

**Comment by lmccart** [comment]
It looks like you are calling createVideo() with a p5.File object, but createVideo() only accepts a string representing the path to a video file (or an array of strings).

**Comment by jsphweid** [comment]
@lmccart Okay, I looked at the source a bit and discovered some things (maybe). 

Ok, so forget the p5.File object, what if the source were simply the data url. That should work, right? Video can be fetched from a data-url: `$('#test').append('<video><source src="' + dataurl + '"></video>');` for example will make use of the data, from what I can tell.

But passing in that into `createVideo(dataurl)`  doesn't seem to work. BUT!!!!! I was throwing debuggers all over the place in and accidentally got it to work. In p5.dom.js, in `function createMedia(`, I put a debugger right below `elt.appendChild(source);` I drag and drop the file and it runs that and stops at the debugger. When I resume execution, the video appears in p5js! But if I take the debugger off, it doesn't load.

Here is an updated plunker demonstrating just that.
https://plnkr.co/edit/0PIMBjxo7Nr2KwkQM1J7

(Be sure to have dev console open so you can hit the debugger in p5.dom)

Seems like some execution is getting a bit too far ahead of itself? What do you think?

**Comment by jsphweid** [comment]
@lmccart @karenpeng Any word on this, my post right before this: loading video from data-url, which seems 'almost' to work?
